### PR TITLE
Adding section on Put_Image aspect

### PR DESCRIPTION
**Pull request by gusthoff**

### CODE REVIEW

File: content/courses/advanced-ada/chapters/strings.rst [hunk 1]
```diff
@@ -856,14 +846,431 @@ Image attribute
...
+attribute. In the code examples from that section, we've seen the default
+behavior of this attribute: the string returned by the calls to :ada:`'Image`
+was always in the format defined by the Ada standard.
+
+In some situations, however, we might want to customize the string that is
+returned by the :ada:`'Image` attribute of a type :ada:`T`. Ada allows us to do
+that via the :ada:`Put_Image` aspect. This is what we have to do:
+
+1. Specify the :ada:`Put_Image` aspect for the type :ada:`T` and indicate a
+   procedure :ada:`P`.
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/strings.rst:871]: Yes, this makes sense.

[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/strings.rst:871]: Thanks!

File: content/courses/advanced-ada/chapters/strings.rst [hunk 2]
```diff
@@ -856,14 +846,431 @@ Image attribute
...
+
+2. Implement the procedure :ada:`P` and write the information we want to use
+   into a buffer (by calling the :ada:`Put` procedure).
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/strings.rst:874]: Yes, I'll add this part.

File: content/courses/advanced-ada/chapters/strings.rst [hunk 3]
```diff
@@ -856,14 +846,431 @@ Image attribute
...
+       Put_Line ("--------------------");
+       Put_Line ("T'Class'Image :   "
+                 & T'Class (Obj_T_Child_1)'Image);
+    end Show_Tagged_Put_Image;
+
+In this example, we declare the type :ada:`T` and its derived types
+:ada:`T_Child_1`, :ada:`T_Child_2` and :ada:`T_Child_3`. When running this
+code, we see that:
+
+- for both :ada:`T_Child_1` and :ada:`T_Child_2` types, the parent's
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/strings.rst:1239]: Yes, agreed.

File: content/courses/advanced-ada/chapters/strings.rst [hunk 4]
```diff
@@ -856,14 +846,431 @@ Image attribute
...
+If the :ada:`Put_Image` aspect of type :ada:`T` isn't specified, the default
+version is used. (We've seen the default version of various types
+:ref:`in the previous section <Adv_Ada_Image_Attribute>` about the :ada:`Image`
+attribute.)
+
+
+:ada:`Put_Image` and derived types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Derived types make use of the :ada:`Put_Image` procedure that was specified for
```
[the remark this answers is not in the document]
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/strings.rst]: Yes, you're right: this could be confusing.

**Review by pat-rogers — DISMISSED** [review]

**Pull request merged**